### PR TITLE
Refactor CUDA operations

### PR DESCRIPTION
**Pull request by davidbeckingsale**

### CODE REVIEW

File: src/umpire/op/CudaAdviseOperation.hpp [hunk 1]
```diff
@@ -7,13 +7,17 @@
 #ifndef UMPIRE_CudaAdviseAccessedByOperation_HPP
 #define UMPIRE_CudaAdviseAccessedByOperation_HPP
```
Comment (mcfadden8) [inline src/umpire/op/CudaAdviseOperation.hpp]: To be consistent with ending comment on line 37, I suggest
```suggestion
#define UMPIRE_CudaAdviseOperation_HPP
```

File: src/umpire/op/CudaAdviseOperation.hpp [hunk 2]
```diff
@@ -7,13 +7,17 @@
 #ifndef UMPIRE_CudaAdviseAccessedByOperation_HPP
```
Comment (mcfadden8) [inline src/umpire/op/CudaAdviseOperation.hpp]: To be consistent with ending comment on line 37, I suggest

```suggestion
#ifndef UMPIRE_CudaAdviseOperation_HPP
```

**Review by mcfadden8 — COMMENTED** [review]
Looks good.  Only had two minor suggestions.

**Review by mcfadden8 — DISMISSED** [review]

**Comment by kab163** [comment]
Maybe edit title so that it notes that these are the cuda (mem?) advise operations that are getting a makeover?

**Comment by davidbeckingsale** [comment]
> Maybe edit title so that it notes that these are the cuda (mem?) advise operations that are getting a makeover?

Copy is also refactored...

**Comment by kab163** [comment]
> > Maybe edit title so that it notes that these are the cuda (mem?) advise operations that are getting a makeover?
> 
> Copy is also refactored...

Ah, ok. I guess I'm not as familiar with these cuda operation funcs, so it wasn't as clear.

**Comment by davidbeckingsale** [comment]
> Ah, ok. I guess I'm not as familiar with these cuda operation funcs, so it wasn't as clear.

This PR exposes the "kind" of copy as a parameter, so we can remove the CopyTo/CopyFrom files

https://github.com/LLNL/Umpire/pull/738/files#diff-c702d4779a558f630ed596d43f49ee87d39c1a6abaaf6d6869404350f21dc40d

**Review by mcfadden8 — DISMISSED** [review]

**Review by mcfadden8 — APPROVED** [review]

**Pull request merged**